src/AutoMapper/Configuration/IMemberConfigurationExpression.cs

```diff
@@ -290,7 +290,9 @@ public interface IProjectionMemberConfiguration<TSource, TDestination, TMember>
     /// <summary>
     /// Ignore this member for LINQ projections unless explicitly expanded during projection
     /// </summary>
-    void ExplicitExpansion();
+    /// <param name="explicitExpansion">Is explicitExpansion active</param>
+    /// <param name="overrideExpansion">Does explicitExpansion can override settings already set by another ExplicitExpansion call</param>
+    void ExplicitExpansion(Boolean explicitExpansion = true, Boolean overrideExpansion = true);
```

Comment:
if i remove overrideExpansion the all thing doesn't work anymore

Reply:
yes i know, but i have a lot of usecase where i would like to have 100 field with ExplicitExpansion and only one Implicit.
Sadly `ForAllMembers` does not permit this. so i thought that i could add the possibility to configure ExplicitExpansion for inheritance and also for disabling only one field.
if i could chain call after ForAllMembers i want have to do this. But because it was explicitly marked as void, i thought it was intended.

Reply:
But you can simply add an `if` in `ForAllMembers` to skip setting `ExpliitExpansion` when you don't need it.

Reply:
> But you can simply add an `if` in `ForAllMembers` to skip setting `ExpliitExpansion` when you don't need it.

What ??!!! I didn't know that i could already do that... (or i didn't understand you correctly)

Reply:
https://github.com/AutoMapper/AutoMapper/blob/104a192407858fe3a54a8a937ca95a34fd7c6959/src/UnitTests/AutoMapperSpecBase.cs#L61

Reply:
i think i was not clear enough... what i was trying to solve is this kind of code :

```
CreateMap<TCheckSet, CheckSetFromCheckRequestDto>()
  .ForMember(dest => dest.Id, m => { m.MapFrom(src => src.Id); m.ExplicitExpansion(); })
  .ForMember(dest => dest.Equipment, m => { m.MapFrom(src => src.Equipment); m.ExplicitExpansion(); })
  .ForMember(dest => dest.PlannedDate, m => { m.MapFrom(src => src.PlannedDate); m.ExplicitExpansion(); })
  .ForMember(dest => dest.IsChecked, m => { m.MapFrom(src => src.IsChecked); m.ExplicitExpansion(); })
  .ForMember(dest => dest.ActualDate, m => { m.MapFrom(src => src.ActualDate); m.ExplicitExpansion(); })
  .ForMember(dest => dest.Status, m => { m.MapFrom(src => src.StatusId); m.ExplicitExpansion(); })
  .ForMember(dest => dest.Comments, m => { m.MapFrom(src => src.Comments); m.ExplicitExpansion(); })
  .ForMember(dest => dest.ActiveTypes, m => { m.MapFrom(src => src.ActiveTypes.Select(p => (TypeOfCheckTypeDto)p.Id)); })
  .ForMember(dest => dest.Jobs, m => { m.MapFrom(src => src.Jobs); m.ExplicitExpansion(); });
//.ForAllMembers(p => p.ExplicitExpansion());

CreateMap<TCheckSet, CheckSetDto>()
  .IncludeBase<TCheckSet, CheckSetFromCheckRequestDto>()
  .ForMember(dest => dest.CheckRequest, m => { m.MapFrom(src => src.CheckRequest); m.ExplicitExpansion(); });
  //.ForAllMembers(p => p.ExplicitExpansion());
```

and would like to write this kind instead

```
CreateMap<TCheckSet, CheckSetFromCheckRequestDto>()
   .ForMember(dest => dest.Id, m => m.MapFrom(src => src.Id))
   .ForMember(dest => dest.Equipment, m => m.MapFrom(src => src.Equipment))
   .ForMember(dest => dest.PlannedDate, m => m.MapFrom(src => src.PlannedDate))
   .ForMember(dest => dest.IsChecked, m => m.MapFrom(src => src.IsChecked))
   .ForMember(dest => dest.ActualDate, m => m.MapFrom(src => src.ActualDate))
   .ForMember(dest => dest.Status, m => m.MapFrom(src => src.StatusId))
   .ForMember(dest => dest.Comments, m => m.MapFrom(src => src.Comments))
   .ForMember(dest => dest.ActiveTypes, m => { m.MapFrom(src => src.ActiveTypes.Select(p => (TypeOfCheckTypeDto)p.Id)); m.ExplicitExpansion(false))
   .ForMember(dest => dest.Jobs, m => m.MapFrom(src => src.Jobs))
   .ForAllMembers(p => p.ExplicitExpansion());

CreateMap<TCheckSet, CheckSetDto>()
    .IncludeBase<TCheckSet, CheckSetFromCheckRequestDto>()
    .ForMember(dest => dest.CheckRequest, m => m.MapFrom(src => src.CheckRequest))
    .ForMember(dest => dest.Jobs, m => { m.MapFrom(src => src.Jobs);  m.ExplicitExpansion(false)})
    .ForAllMembers(p => p.ExplicitExpansion());
```

notice the subtle change for the ActiveTypes field and the use of ForAllMembers (and of course now the possibility to disable ExplicitExpansion for some fields) 
but actually, if i'm doing this the settings of ActiveTypes is reset when i call ForAllMembers (witch the inheritance too, which could be more problematic)

Reply:
i could even add , in this case i could rewrite the all thing like this

```
CreateMap<TCheckSet, CheckSetFromCheckRequestDto>()
   .ForMember(dest => dest.ActiveTypes, m => { m.MapFrom(src => src.ActiveTypes.Select(p => (TypeOfCheckTypeDto)p.Id)); m.ExplicitExpansion(false))
   .ForAllMembers(p => p.ExplicitExpansion());

CreateMap<TCheckSet, CheckSetDto>()
    .IncludeBase<TCheckSet, CheckSetFromCheckRequestDto>()
    .ForMember(dest => dest.CheckRequest, m => m.MapFrom(src => src.CheckRequest))
    .ForMember(dest => dest.Jobs, m => { m.MapFrom(src => src.Jobs);  m.ExplicitExpansion(false)})
    .ForAllMembers(p => p.ExplicitExpansion());
```

which feel really clean.. instead of all the config lines